srml/staking/src/lib.rs

```diff
@@ -398,6 +398,9 @@ pub trait Trait: system::Trait + session::Trait {
 	/// The staking balance.
 	type Currency: LockableCurrency<Self::AccountId, Moment=Self::BlockNumber>;
 
+	/// The balance of an account. Redefined with additional trait bound.
+	type Balance: As<<Self::Currency as Currency<<Self as system::Trait>::AccountId>>::Balance> + As<usize>;
```

Comment:
If the PR now is about some refactor of As bound. Then this is not needed as validators len is usize, so it can be converted to u64 safely (no ?) and so we can use `As<u64>` of simple arithmetic.

I'm still not sure about the final direction for the future but I think it will use `TryFrom<u64>` bounds in simplearithmetic.

(by the way It seems to me we should be able to convert from a small enough integer representation to simple arithmetic (or at least currency) as we could do `for _ in 0..len { value += one() }`)